frontstage/controllers/oauth_controller.py

```diff
@@ -0,0 +1,44 @@
+import json
+import logging
+
+from flask import current_app as app
+import requests
+from structlog import wrap_logger
+
+from frontstage.exceptions.exceptions import ApiError, OAuth2Error
+
+
+logger = wrap_logger(logging.getLogger(__name__))
+
+
+def sign_in(username, password):
+    logger.debug('Attempting to retrieve OAuth2 token for sign-in')
+    
+    url = f"{app.config['OAUTH_SERVICE_URL']}/api/v1/tokens/"
+    data = {
+        'grant_type': 'password',
+        'client_id': app.config['OAUTH_CLIENT_ID'],
+        'client_secret': app.config['OAUTH_CLIENT_SECRET'],
+        'username': username,
+        'password': password,
+    }
+    headers = {
+        'Accept': 'application/json',
+        'Content-Type': 'application/x-www-form-urlencoded;charset=UTF-8',
+    }
+    response = requests.post(url, headers=headers, auth=app.config['OAUTH_BASIC_AUTH'], data=data)
```

Comment:
Are network level errors handled by the application, or should they be handled here?

Reply:
Yup - [error_handlers.py](https://github.com/ONSdigital/ras-frontstage/blob/master/frontstage/error_handlers.py#L22).

Reply:
@JamesGardiner should be handled by the [error_handlers.py](https://github.com/ONSdigital/ras-frontstage/blob/master/frontstage/error_handlers.py#L30).